### PR TITLE
Add a script to update dependencies for tests and normalize KN and OCP versions

### PR DESCRIPTION
**Pull request by markusthoemmes**
As per title, since we're moving to go.mod almost everywhere, I think it makes sense to have an `update-deps.sh` script like upstream has.

I took the opportunity to align the dependencies to the upcoming 0.17 release. In the same go, I pulled the K8s deps to 0.17 too (what Knative has) and made an effort to normalize OCP related dependencies on a common version as well. I hope this simplifies version bumps in the future and I'm looking forward to getting to 0.18 as that'll hopefully allow us to align the OCP dependencies as well.

Currently I had to choose OCP4.4 dependencies because 4.5 needs 0.18 K8s dependencies, which had breaking API changes. Those will come with a Knative bump to 0.18.

### CODE REVIEW

**Comment by markusthoemmes** [comment]
/hold

This looks like it might be relevant. Gonna have to scrutinize how we run the tests to be sure.

**Comment by markusthoemmes** [comment]
/test 4.5-upstream-e2e-aws-ocp-45

To see if this is consistent. This would only be related if somehow our scripts are calling into the wrong codebase :thinking:

**Comment by markusthoemmes** [comment]
/retest

Known issues over night.

**Comment by markusthoemmes** [comment]
/test 4.5-upgrade-tests-aws-ocp-45

Filed https://issues.redhat.com/browse/SRVKS-644 to squash upgrade instability.

**Comment by markusthoemmes** [comment]
/unhold

This is fine now.

File: test/go.mod [hunk 1]
```diff
@@ -3,50 +3,49 @@ module github.com/openshift-knative/serverless-operator/test
...
-	knative.dev/eventing v0.15.3
-	knative.dev/operator v0.14.3-0.20200605190218-43fce8e820ed
-	knative.dev/pkg v0.0.0-20200625173728-dfb81cf04a7c
-	knative.dev/serving v0.15.3
-	knative.dev/test-infra v0.0.0-20200519161858-554a95a37986
+	knative.dev/eventing v0.17.6-0.20201006114020-60caedca1325
+	knative.dev/operator v0.17.2
+	knative.dev/pkg v0.0.0-20200831162708-14fb2347fb77
+	knative.dev/serving v0.17.3
+	knative.dev/test-infra v0.0.0-20200915193842-f4d4232c1f04
```
Comment (matzew) [inline test/go.mod:18]: +1

**Review by matzew — APPROVED** [review]
thanks for version/dep bump for most of the deps, and adding the update script

/lgtm
/approve

**Pull request merged**